src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2095,6 +2095,45 @@ def aead_cipher_supported(self, cipher):
             self._lib.EVP_get_cipherbyname(cipher_name) != self._ffi.NULL
         )
 
+    def load_pkcs12(self, data, password):
+        if password is None:
+            password = self._ffi.NULL
+        elif not isinstance(password, bytes):
+            raise TypeError("Password must be a byte string or None")
+
+        bio = self._bytes_to_bio(data)
+        p12 = self._lib.d2i_PKCS12_bio(bio.bio, self._ffi.NULL)
+        if p12 == self._ffi.NULL:
+            raise ValueError("Could not deserialize PKCS12 data")
+
+        p12 = self._ffi.gc(p12, self._lib.PKCS12_free)
+        evp_pkey_ptr = self._ffi.new("EVP_PKEY **")
+        x509_ptr = self._ffi.new("X509 **")
+        res = self._lib.PKCS12_parse(
+            p12, password, evp_pkey_ptr, x509_ptr, self._ffi.NULL
+        )
+        if res == 0:
+            errors = self._consume_errors()
+            # TODO: Chances are there are more errors we can cause here, but
+            # we currently have no test vectors that can generate that failure
+            # type.
+            self.openssl_assert(
```

Comment:
Do we even want to error in this fashion? BoringSSL doesn't have this error code, so I'm tempted to say we should just consume the errors and just raise a ValueError saying we couldn't deserialize and that it MIGHT be a bad password.